### ISSUE TITLE
"Panic: uncaught pattern violation" involving new `with ... in`

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda#5818]
Another bug I found when trying to push through https://github.com/agda/agda-stdlib/pull/1630.
```agda
open import Agda.Builtin.Nat using (Nat)
open import Agda.Builtin.Equality using (refl)

foo : Nat → Nat
foo i with refl in eq
... | refl = i
```
If you remove `in eq` then the error goes away.

**Comment by lawcho** [comment]
The Sigma type causes this issue too:
```.agda

open import Agda.Builtin.Sigma
open import Agda.Builtin.Unit
open import Agda.Builtin.Equality

mk-con : Σ ⊤ λ  _ → ⊤
mk-con = tt , tt

test : ⊤
test with mk-con in eq
... | z = tt

```
As before, removing `in eq` makes the error go away.

(tested on Agda 2.6.2.1)

**Comment by andreasabel** [comment]
According to my experiments, this is fixed on 2.6.2.2.  You get unsolved constraints now (no uncaught pattern violation).

Please reopen if the behavior of 2.6.2.2 does not match your expectations.
